### PR TITLE
Slick Moves: Various improvements to local setup

### PR DESCRIPTION
**Pull request by Shadowfiend**
The highlights:
- Using `npx truffle` to remove the global truffle requirement. This has some dependent PRs that need to be merged first, since it also hits install scripts in other repos: keep-network/keep-core#2021, keep-network/keep-ecdsa#538, keep-network/tbtc#727.
- Updating the Electrum config suggested in manual-setup for the dApp; the updated code depends on keep-network/tbtc.js#81. This will be obsoleted by a tbtc-dapp PR that handles the local network directly, keep-network/tbtc-dapp#356, but which does not block this PR.
- Move geth to Clique PoA. This should significantly reduce CPU load for the local setup (once those pesky primes are generated).
- Move geth to a system-designated port. This handles ethereum/go-ethereum#1271, which I ran into.

Drafted until upstream PRs are merged.

### CODE REVIEW

**Comment by Shadowfiend** [comment]
Once the upstreams are merged, I'm also going to bump this to latest `master`.

Also, I was thinking about switching the submodules to use [branch tracking for `master`](https://git-scm.com/docs/git-submodule/en#Documentation/git-submodule.txt--bltbranchgt). Thoughts? Basically that means whenever you `git submodule update` you pull the latest version of that branch from the remote, rather than being fixed on one commit.

**Comment by lukasz-zimnoch** [comment]
> Once the upstreams are merged, I'm also going to bump this to latest `master`.
> 
> Also, I was thinking about switching the submodules to use [branch tracking for `master`](https://git-scm.com/docs/git-submodule/en#Documentation/git-submodule.txt--bltbranchgt). Thoughts? Basically that means whenever you `git submodule update` you pull the latest version of that branch from the remote, rather than being fixed on one commit.

Yeah, good point. We are already doing this but only during the installation in https://github.com/keep-network/local-setup/blob/master/install-repositories.sh

**Comment by Shadowfiend** [comment]
Actually I misread how `--remote` works. We can leave the branch off, since by doing that I believe we get the remote default branch anyway.

File: docs/manual-setup.adoc [hunk 1]
```diff
@@ -8,7 +8,6 @@ If you are not on macOS or you don't have Homebrew, you need to install:
 - Go compiler, at least 1.14
 - Geth (Ethereum client) 1.9.9-stable
 - Solidity, at least 0.5.17
-- Truffle, at least 5.1.18
```
Comment (lukasz-zimnoch) [inline docs/manual-setup.adoc:11]: I think we should do the same for the auto deployment and remove truffle from Packer provisioning scripts.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**